lib/gssapi/mech/gss_add_cred_from.c

```diff
@@ -88,14 +88,75 @@ _gss_mg_add_mech_cred(OM_uint32 *minor_status,
     } else
 	major_status = GSS_S_UNAVAILABLE;
 
-    if (major_status == GSS_S_COMPLETE && out)
```

Comment:
But we don't want to return an error, nor call the mechanism's acquisition methods, if the `desired_mech` is a `GM_USE_MG_CRED` mechanism.

Reply:
That can't happen. `add_mech_cred_internal()`, which calls the function you commented above, is always called with a concrete mechanism.

Reply:
There's even an assertion to validate this :)

Reply:
However: if `desired_mech` is a `GM_USE_MG_CRED` mechanism, are you saying we should make it a NOOP rather than adding creds for all mechanisms?

Reply:
This we resolved on IM last night with proposed new code from @lhoward.